### ISSUE TITLE
Consider providing ES6 builds

### ISSUE THREAD

**Issue by wojtekmaj** [launchdarkly/js-client-sdk#254]
**Is your feature request related to a problem? Please describe.**

Currently, the code is compiled to ES5. This makes js-client-sdk way heavier than it needs to be. From my experiments, as much as 1/4 could be shaved off by compiling code to ES6.

**Describe the solution you'd like**

`target: "es6"` :)

**Comment by kinyoklion** [comment]
Hello @wojtekmaj,

Thank you for the request.

The main interest here is the packaged size of the SDK, correct?

_Filed internally as 162092_

Thank you,
Ryan

**Comment by wojtekmaj** [comment]
Yep! You technically can use e.g. Webpack alias to rewrite imports to use src folder & build the library yourself, but I think it's too much of a hassle for most developers. Even having two builds alongside, if you don't want any breaking changes for the time being, would be a big quality of life improvement.

**Comment by vitalikda** [comment]
@kinyoklion any updates here? Looks like react sdk was already converted to ES6

**Comment by adrigardi90** [comment]
Same, it would be helpful to get it compiled to ES6 so the bundle size improves. I would have expected it to be smaller in the first place

**Comment by tniezurawski** [comment]
Looks like the size of the lib is growing:

![image](https://github.com/launchdarkly/js-client-sdk/assets/4335166/a9b8a864-1efb-423a-b1f3-12cf920763a2)

As mentioned above, react SDK benefited from the switch:

![image](https://github.com/launchdarkly/js-client-sdk/assets/4335166/aa75db4e-71f1-49c4-bc62-c69e2a7a8999)

although the numbers are not as impressive as @wojtekmaj's predictions maybe the result would be better for this project 🤷‍♂️ . Anyway, maybe compiling to ES6 is not a bad idea?
